src/modules/AutoinstConfig.rb

```diff
@@ -214,114 +219,157 @@ def setProposalList(l)
-                  #        so it's lazy but okay. No reason to be too strict here with the checks
-                  #        As soon as more than one attr is possible, I need to iterate over the attr list
-                  #
-                  comment = Ops.get(attrList, 0, "")
-                  # The line above needs to be fixed when we have more attributes
-
-                  # comment will look like this: "(description=BLA BLA)"
-                  startComment = Builtins.findfirstof(comment, "=")
-                  endComment = Builtins.findlastof(comment, ")")
-                  if startComment != nil && endComment != nil &&
-                      Ops.greater_than(
-                        Ops.subtract(Ops.subtract(endComment, startComment), 1),
-                        0
-                      )
-                    comment = Builtins.substring(
-                      comment,
-                      Ops.add(startComment, 1),
-                      Ops.subtract(Ops.subtract(endComment, startComment), 1)
-                    )
-                  else
-                    comment = ""
-                  end
-                  if Ops.less_than(Builtins.size(comment), 1)
-                    comment = Builtins.sformat(
-                      "bad description in SLP for %1",
-                      url
-                    )
-                  end
-                  dummy = Builtins.add(dummy, Item(comment, false))
-                  Ops.set(comment2url, comment, url)
-                else
-                  url = Builtins.substring(Ops.get_string(m, "srvurl", ""), 17)
-                  dummy = Builtins.add(dummy, Item(url, false))
-                  Ops.set(comment2url, url, url)
-                end
-              end
-              dlg = Left(ComboBox(Id(:choose), _("Choose Profile"), dummy))
-              UI.OpenDialog(VBox(dlg, PushButton(Id(:ok), Label.OKButton)))
-              UI.UserInput
-              cmdLine = Ops.get(
-                comment2url,
-                Convert.to_string(UI.QueryWidget(Id(:choose), :Value)),
-                ""
+    # Searches for 'autoyast' via SLP and returns the full URL of
+    # the profile. If more providers are found, user is asked to
+    # select one.
+    #
+    # FIXME: This function has been intentionally left (almost) intact
+    # and needs refactoring
+    #
+    # @return [String] profile location or 'nil' if nothing is found
+    def find_slp_autoyast
+      profile_location = nil
+
+      slpData = SLP.FindSrvs("autoyast", "")
+
+      # More providers to choose from
+      if Ops.greater_than(Builtins.size(slpData), 1)
+        dummy = []
+        comment2url = {}
+        Builtins.foreach(slpData) do |m|
+          attrList = SLP.FindAttrs(Ops.get_string(m, "srvurl", ""))
+
+          if Ops.greater_than(Builtins.size(attrList), 0)
+            url = Builtins.substring(Ops.get_string(m, "srvurl", ""), 17)
```

Comment:
Yes, that's the right question :)

because `srvurl` is, e.g., `service:autoyast:http://server.example.org/autoinst.xml`